onnx/common/ir.h

```diff
@@ -1138,6 +1136,41 @@ friend struct Value;
 
   friend std::ostream& operator<<(std::ostream & out, const Graph & g);
 
+  void forSelfAndEachSubGraph(std::function<void(Graph*)> fn) {
+    fn(this);
+
+    for (const Node* node : all_nodes) {
+      for (const auto& attr : node->attributeNames()) {
+        if (node->kindOf(attr) == AttributeKind::g) {
+          std::shared_ptr<Graph> subgraph = node->g(attr);
+          subgraph->forSelfAndEachSubGraph(fn);
+        } else if (node->kindOf(attr) == AttributeKind::gs) {
+          for (const auto& subgraph : node->gs(attr)) {
+            subgraph->forSelfAndEachSubGraph(fn);
+          }
+        }
+      }
+    }
+  }
+
+  void forSelfAndEachSubGraph(std::function<void(const Graph*)> fn) const {
```

Comment:
I am confused why do you need a const and a non const version of these methods? cant the invoker const cast and directly call the non const version?

Reply:
I don't think it is a good habit to do const_cast and call the non const version.
For the const `forSelfAndEachSubGraph` method, the argument of `fn` is `const Graph`, that means `fn` cannot (and should not) modify the graph. For the non-const version there is not such a constraint